deepinterpolation/generator_collection.py

```diff
@@ -536,126 +398,163 @@ def __data_generation__(self, local_x, local_y, local_z, local_t):
 
-        shape = (self.total_frame_per_movie, int(self.nb_probes / 2), 2)
-        # load it with the correct shape
-        self.raw_data = np.memmap(
-            self.raw_data_file, dtype="int16", shape=shape)
+        if (self.end_sample - self.start_sample+1) < self.batch_size:
+            raise Exception("Not enough frames to construct one " +
+                            str(self.batch_size) + " frame(s) batch between " +
+                            str(self.start_sample) +
+                            " and "+str(self.end_sample) +
+                            " frame number.")
 
-        # Older reshape code, to remove when stable
-        # Reshape in number of traces
-        # self.raw_data = np.reshape(self.raw_data
-        #   , (self.total_frame_per_movie,
-        #   self.nb_probes))
+        # +1 to make sure end_samples is included
+        self.list_samples = np.arange(self.start_sample, self.end_sample+1)
 
-        # Reshape following probes location
-        # self.raw_data = np.reshape(self.raw_data, (self.total_frame_per_movie
-        #                                           int(self.nb_probes/2), 2)
-
-        local_data = self.raw_data[0:average_nb_samples, :, :].flatten()
-        local_data = local_data.astype("float32")
-        self.local_mean = np.mean(local_data)
-        self.local_std = np.std(local_data)
-        self.epoch_index = 0
-        self.list_samples = np.arange(
-            self.start_frame, self.start_frame + self.img_per_movie
-        )
-        if "randomize" in self.json_data.keys():
-            if self.json_data["randomize"] == 1:
-                np.random.shuffle(self.list_samples)
+        if self.randomize:
+            np.random.shuffle(self.list_samples)
 
         # We cut the number of samples if asked to
         if (self.total_samples > 0
                 and self.total_samples < len(self.list_samples)):
             self.list_samples = self.list_samples[0: self.total_samples]
 
-    def __len__(self):
-        "Denotes the total number of batches"
-        return int(np.floor(float(len(self.list_samples)) / self.batch_size))
-
     def on_epoch_end(self):
-        # We only increase index if steps_per_epoch
-        # is set to positive value. -1 will force the generator
-        # to not iterate at the end of each epoch
+        """We only increase index if steps_per_epoch is set to positive value.
+        -1 will force the generator to not iterate at the end of each epoch."""
         if self.steps_per_epoch > 0:
             if self.steps_per_epoch * (self.epoch_index + 2) < self.__len__():
                 self.epoch_index = self.epoch_index + 1
             else:
                 # if we reach the end of the data, we roll over
                 self.epoch_index = 0
 
-    def __getitem__(self, index):
+    def __len__(self):
+        "Denotes the total number of batches"
+        return int(np.floor(float(len(self.list_samples)) / self.batch_size))
```

Comment:
`float` is not needed in python3 since division returns a float. `np.floor` is not needed. `int` does the same thing as `np.floor`

Reply:
Yes, I was suggesting to replace `np.floor` with `int`

Reply:
Sorry

Reply:
True fixed